components/Starknet/modules/architecture-and-concepts/pages/smart-contracts/system-calls-cairo1.adoc

```diff
@@ -463,3 +463,68 @@ None.
 .Common library
 
 link:https://github.com/starkware-libs/cairo/blob/cca08c898f0eb3e58797674f20994df0ba641983/corelib/src/starknet/syscalls.cairo#L70[`syscalls.cairo`^]
+
+[id="keccak"]
+== `keccak`
+
+[discrete]
+=== Function signature
+
+[source,cairo,subs="+quotes,+macros"]
+----
+extern fn keccak_syscall(
+    data: Span<felt252>
+) -> SyscallResult<u256> implicits(GasBuiltin, System) nopanic;
+----
+
+[discrete]
+=== Description
+
+Computes the Keccak-256 hash of the input data. This system call is particularly useful when interacting with Ethereum contracts or implementing Ethereum-compatible functionality, as Keccak-256 is widely used in the Ethereum ecosystem.
+
+[discrete]
+=== Arguments
+
+[horizontal,labelwidth=35]
+`_data_: Span<felt252>`:: The input data to be hashed, represented as an array of field elements.
+
+[discrete]
+=== Return values
+
+* The Keccak-256 hash of the input data as a `u256` value, wrapped in `SyscallResult`.
+
+[discrete]
+=== Common library
+
+link:https://github.com/starkware-libs/cairo/blob/cca08c898f0eb3e58797674f20994df0ba641983/corelib/src/starknet/syscalls.cairo#L84[`syscalls.cairo`^]
+
+[discrete]
+=== Example
```

Comment:
The way to use this syscall is not directly, but with functions in `[keccak.cairo](https://github.com/starkware-libs/cairo/blob/67c6eff9c276d11bd1cc903d7a3981d8d0eb2fa2/corelib/src/keccak.cairo)` that themselves use the syscall under the hood.
Please change the example accordingly and add an explanation under "Description".